examples/vision/shiftvit.py

```diff
@@ -0,0 +1,866 @@
+        initial_filters = int(self.mlp_expand_ratio * input_channels)
+
+        self.mlp = keras.Sequential(
+            [
+                layers.Conv2D(
+                    filters=initial_filters,
+                    kernel_size=1,
+                    padding="same",
+                    activation=tf.nn.gelu,
+                ),
+                layers.Dropout(rate=self.mlp_dropout_rate),
+                layers.Conv2D(filters=input_channels, kernel_size=1, padding="same"),
+                layers.Dropout(rate=self.mlp_dropout_rate),
+            ]
+        )
+
+    def call(self, x):
+        x = self.mlp(x)
+        return x
+
+
+"""
+#### Drop Path
+
+Stochastic depth is a regularization technique that randomly drops a set of
+layers. During inference, the layers are kept as they are. It is very much
+similar to Dropout but only that it operates on a block of layers rather
+than individual nodes present inside a layer.
+"""
+
+
+class DropPath(layers.Layer):
+    """Drop Path also known as the Stochastic Depth layer.
+
+    Refernece:
+        - https://keras.io/examples/vision/cct/#stochastic-depth-for-regularization
+        - github.com:rwightman/pytorch-image-models
+    """
+
+    def __init__(self, drop_path_prob, **kwargs):
+        super().__init__(**kwargs)
+        self.drop_path_prob = drop_path_prob
+
+    def call(self, x, training=False):
+        if training:
+            keep_prob = 1 - self.drop_path_prob
+            shape = (tf.shape(x)[0],) + (1,) * (len(tf.shape(x)) - 1)
+            random_tensor = keep_prob + tf.random.uniform(shape, 0, 1)
+            random_tensor = tf.floor(random_tensor)
+            return (x / keep_prob) * random_tensor
+        return x
+
+
+"""
+#### Block
+
+The most important operation in this paper is the **shift opperation**. In this section,
+we describe the shift operation, and also compare it with the original implementation (in
+PyTorch).
+
+A generic feature map is assumed to be of the shape `[N, H, W, C]`. Here we choose a
+`num_div` parameter that decides the division size of the channels. The first 4 divisions
+are shifted (1 pixel) in the left, right, up, and down direction. The remaining splits
+are kept as is. After partial shifting the shifted channels are padded and the overflown
+pixels are chopped off. This completes the partial shifting operation.
+
+In the original implementation the code is something like
+```python
+out[:, g * 0:g * 1, :, :-1] = x[:, g * 0:g * 1, :, 1:]  # shift left
+out[:, g * 1:g * 2, :, 1:] = x[:, g * 1:g * 2, :, :-1]  # shift right
+out[:, g * 2:g * 3, :-1, :] = x[:, g * 2:g * 3, 1:, :]  # shift up
+out[:, g * 3:g * 4, 1:, :] = x[:, g * 3:g * 4, :-1, :]  # shift down
+
+out[:, g * 4:, :, :] = x[:, g * 4:, :, :]  # no shift
+```
+
+In TensorFlow it would be infeasible for us to assign shifted channels to a tensor in the
+middle of a training process. This is why we have resorted to the following procedure:
```

Comment:
There are several things that could be done to implement just this, but concatenation works fine as well.

Reply:
What would be the best way according to you? This would just be a great learning curve for me.

Reply:
None of the alternative ways are super satisfactory to be honest. You could use `tensor_scatter_nd_add `, you could use `tf.experimental.numpy`, you could all the XLA op `dynamic_update_slice` directly...

Reply:
Makes sense to me! Thanks.